packages/odie-client/src/components/message/messages-container.tsx

```diff
@@ -46,175 +46,197 @@ interface ChatMessagesProps {
-	 */
-	useEffect( () => {
-		if (
-			isForwardingToZendesk &&
-			! hasForwardedToZendesk &&
-			! chat.conversationId &&
-			createZendeskConversation &&
-			resetSupportInteraction &&
-			isChatLoaded
-		) {
-			searchParams.delete( 'provider' );
-			searchParams.set( 'direct-zd-chat', '1' );
-			setSearchParams( searchParams );
-			setHasForwardedToZendesk( true );
-
-			// when forwarding to zd avoid creating new chats
-			if ( alreadyHasActiveZendeskChat ) {
+export const MessagesContainer = forwardRef(
+	( { currentUser }: ChatMessagesProps, forwardedRef: ForwardedRef< HTMLDivElement > ) => {
+		const { chat, botNameSlug, isChatLoaded, isUserEligibleForPaidSupport } =
+			useOdieAssistantContext();
+		const sendMessage = useSendChatMessage();
+		const createZendeskConversation = useCreateZendeskConversation();
+		const resetSupportInteraction = useResetSupportInteraction();
+		const [ searchParams, setSearchParams ] = useSearchParams();
+		const isForwardingToZendesk =
+			searchParams.get( 'provider' ) === 'zendesk' && chat.provider !== 'zendesk';
+		const [ hasForwardedToZendesk, setHasForwardedToZendesk ] = useState( false );
+		const [ chatMessagesLoaded, setChatMessagesLoaded ] = useState( false );
+		const [ shouldEnableAutoScroll, setShouldEnableAutoScroll ] = useState( true );
+		const navType: NavigationType = useNavigationType();
+
+		const scrollParentRef = useRef< HTMLElement | null >( null );
+
+		const { alreadyHasActiveZendeskChat, chatHasEnded } = useSelect( ( select ) => {
+			const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+			const currentInteraction = helpCenterSelect.getCurrentSupportInteraction();
+			return {
+				alreadyHasActiveZendeskChat:
+					interactionHasZendeskEvent( currentInteraction ) &&
+					! interactionHasEnded( currentInteraction ),
+				chatHasEnded: interactionHasEnded( currentInteraction ),
+			};
+		}, [] );
+
+		const lastMessage = chat.messages?.[ chat.messages.length - 1 ];
+		const predictions = lastMessage?.predictions;
+
+		const handlePredictionClick = useCallback(
+			( prediction: string, predictionKey: string ) => {
+				sendMessage( { content: prediction, role: 'user', type: 'message', predictionKey } );
+			},
+			[ sendMessage ]
+		);
+
+		useZendeskMessageListener();
+		useAutoScroll( forwardedRef, shouldEnableAutoScroll );
+		useHelpCenterChatScroll(
+			chat?.supportInteractionId,
+			scrollParentRef,
+			! shouldEnableAutoScroll
+		);
+
+		useEffect( () => {
+			if ( navType === 'POP' && ( isChatLoaded || ! isUserEligibleForPaidSupport ) ) {
+				setShouldEnableAutoScroll( false );
+			}
+		}, [ navType, isUserEligibleForPaidSupport, shouldEnableAutoScroll, isChatLoaded ] );
+
+		useEffect( () => {
+			if ( forwardedRef && scrollParentRef.current === null ) {
+				scrollParentRef.current =
+					( forwardedRef as React.RefObject< HTMLDivElement > ).current?.closest(
+						'.help-center__container-content'
+					) ?? null;
+			}
+		}, [ forwardedRef ] );
```

Comment:
I think that putting `forwardedRef` in the dependency array won't cause the effect to re-run, since it should always return the same object.